src/test/java/seedu/address/logic/commands/AddAppointmentCommandTest.java

```diff
@@ -0,0 +1,291 @@
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalPersons.BENSON_P;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.AppointmentBook;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.ReadOnlyAppointmentBook;
+import seedu.address.model.ReadOnlyUserPrefs;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.AppointmentDescriptor;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.PersonDescriptor;
+import seedu.address.testutil.AppointmentBuilder;
+
+public class AddAppointmentCommandTest {
+    @Test
+    public void constructor_nullPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddAppointmentCommand(null, 1));
+    }
+
+    @Test
+    public void execute_appointmentAcceptedByModel_addSuccessful() throws Exception {
+        ModelStubAcceptingAppointmentAdded modelStub = new ModelStubAcceptingAppointmentAdded();
+        AppointmentDescriptor validAppointementDescriptor = new AppointmentBuilder().build().getAppointmentDescriptor();
+
+        CommandResult commandResult = new AddAppointmentCommand(validAppointementDescriptor, 1)
+                .execute(modelStub);
+
+        assertEquals(String.format(AddAppointmentCommand.MESSAGE_SUCCESS,
+                Messages.formatAppointment(validAppointementDescriptor)), commandResult.getFeedbackToUser());
+        assertEquals(List.of(validAppointementDescriptor), modelStub.appointmentsAdded);
+    }
+
+    @Test
+    public void execute_duplicatePerson_throwsCommandException() {
+        AppointmentDescriptor validAppointmentDescriptor = new AppointmentBuilder().build().getAppointmentDescriptor();
+        Appointment validAppointment = new AppointmentBuilder().build();
+        AddAppointmentCommand addAppointmentCommand = new AddAppointmentCommand(validAppointmentDescriptor, 1);
+        AddAppointmentCommandTest.ModelStub modelStub = new AddAppointmentCommandTest
+                .ModelStubWithAppointment(validAppointment);
+
+        assertThrows(CommandException.class,
+                AddAppointmentCommand.MESSAGE_DUPLICATE_APPOINTMENT, () -> addAppointmentCommand.execute(modelStub));
+    }
+
+    @Test
+    public void execute_getPersonIdDoesNotExists_returnsCorrectMessage() {
+        AppointmentDescriptor validAppointmentDescriptor = new AppointmentBuilder().build().getAppointmentDescriptor();
+        Appointment validAppointment = new AppointmentBuilder().build();
```

Comment:
Should line 67 be removed since `validAppointment` is not used in the testing?

Reply:
Removed the line